.github/workflows/shellcheck.yml

```
@@ -0,0 +1,17 @@
+name: shellcheck
+
+on:
+  pull_request:
+
+jobs:
+  shellcheck:
+    name: shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
```

```suggestion
      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
```

I didn't see a reason not to upgrade to v3, project seems to use up to date runners so [v3 requirements](https://github.com/actions/checkout/releases/tag/v3.0.0) should not be a problem.

Otherwise we can upgrade to: `7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2.4.2`

---

done